src/components/AnnouncementsModal.js

```diff
@@ -0,0 +1,246 @@
+            : getValueForAnnouncementItem('hoverBackgroundColor')};
+        }
+      `}
+      onClick={() => {
+        onClick(id);
+      }}
+    >
+      {status === AnnouncementItemStatus.UNSEEN ? (
+        <Icon name={Icons.EYE} />
+      ) : (
+        <Icon name={Icons.EYE_SLASH} />
+      )}
+      <Box
+        as="span"
+        css={`
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+        `}
+      >
+        {title}
+      </Box>
+    </ListItem>
+  );
+};
+
+AnnouncementItem.propTypes = {
+  id: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  status: PropTypes.oneOf(Object.values(AnnouncementItemStatus)),
+  onClick: PropTypes.func,
+};
+
+AnnouncementItem.defaultProps = {
+  status: AnnouncementItemStatus.UNSEEN,
+};
+
+const AnnouncementContent = ({
+  title,
+  imageSrc,
+  body,
+  callToAction,
+  callToActionLabel,
+}) => (
+  <Stack
+    forwardedAs="article"
+    padding={4}
+    spacing={3}
+    css={`
+      width: 70%;
+    `}
+  >
+    <Title>{title}</Title>
+    {imageSrc && (
+      <Link href={callToAction}>
+        <Image
+          src={imageSrc}
+          alt={callToActionLabel}
+          css={`
+            width: 100%;
+            max-height: 30vh;
+            background-position: center center;
+            background-repeat: no-repeat;
+            object-fit: cover;
+
+            &:hover {
+              opacity: 0.85;
+            }
+          `}
+        />
+      </Link>
+    )}
+
+    <div
+      dangerouslySetInnerHTML={{ __html: body }}
+      css={`
+        strong {
+          font-weight: bold;
+        }
+
+        a {
+          color: ${getValueForAnnouncementContent('linkColor')};
+        }
+
+        p {
+          margin-bottom: ${parseSpacing(4)}rem;
+        }
+
+        ol {
+          list-style-type: decimal;
+          margin-bottom: ${parseSpacing(4)}rem;
+
+          li {
+            margin-left: ${parseSpacing(5)}rem;
+          }
+        }
+        ul {
+          list-style-type: disc;
+          margin-bottom: ${parseSpacing(4)}rem;
+
+          li {
+            margin-left: ${parseSpacing(5)}rem;
+          }
+        }
+      `}
+    />
+    <Inline as="div" justifyContent="flex-end">
+      {callToAction && (
+        <Link href={callToAction}>
+          <Button>{callToActionLabel}</Button>
+        </Link>
+      )}
+    </Inline>
+  </Stack>
+);
+
+AnnouncementContent.propTypes = {
+  title: PropTypes.string,
+  imageSrc: PropTypes.string,
+  body: PropTypes.node,
+  callToAction: PropTypes.string,
+  callToActionLabel: PropTypes.string,
+};
+
+const AnnouncementsModal = ({
+  visible,
+  announcements,
+  onClickAnnouncement,
+  onShow,
+  onClose,
+}) => {
+  const { t } = useTranslation();
+
+  const activeAnnouncement = () => {
+    return announcements.find(
+      (announcement) => announcement.status === AnnouncementItemStatus.ACTIVE,
+    );
+  };
```

Comment:
Don't think this needs to be a function, no?

Reply:
Yo have to filter the announcement that has status 'active' to show the content of, no?

Reply:
Sure, but not as a function?